kayenta-prometheus/src/main/java/com/netflix/kayenta/prometheus/config/PrometheusResponseConverter.java

```diff
@@ -73,7 +73,22 @@ public Object fromBody(TypedInput body, Type type) throws ConversionException {
           List<Double> dataValues = new ArrayList<Double>(values.size());
 
           for (List tuple : values) {
-            dataValues.add(Double.valueOf((String) tuple.get(1)));
+            String val = (String) tuple.get(1);
+            if (val != null) {
+              switch (val) {
+                case "+Inf":
+                  dataValues.add(Double.POSITIVE_INFINITY);
+                  break;
+                case "-Inf":
+                  dataValues.add(Double.NEGATIVE_INFINITY);
+                  break;
+                case "NaN":
+                  dataValues.add(Double.NaN);
+                  break;
```

Comment:
It looks like [`Double.valueOf` should handle this case already](https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html#valueOf-java.lang.String-), although perhaps it's more clear to spell it out here anyway? It wants "Infinity" instead of "Inf", unfortunately.

Reply:
Yeah. I thought about it, but decided to add it in case it they change it in Java in the future for any reason. It's like, you either track how NaN is spelled by Prometheus and at the same time ensure that Java parser expects the same, or you just map it explicitly to the constant of the Double class.